src/fdc/DiskManipulator.cc

```diff
@@ -403,17 +403,14 @@ void DiskManipulator::create(std::span<const TclObject> tokens)
 	for (const auto& token_ : view::drop(tokens, 3)) {
 		if (auto t = parseBootSectorType(token_.getString())) {
 			bootType = *t;
-			continue;
-		}
-
-		if (sizes.size() >= MAX_PARTITIONS) {
-			throw CommandException(
-				"Maximum number of partitions is ", MAX_PARTITIONS);
+		} else if (sizes.size() >= MAX_PARTITIONS) {
+			throw CommandException("Maximum number of partitions is ", MAX_PARTITIONS);
+		} else if (size_t sectors = parseSectorSize(token_.getString()); sectors <= ~0u) {
```

Comment:
This is correct, and compact, though I prefer a more direct way (I took me a few moments to realize what was going on here):
```
} else if (size_t sectors = parseSectorSize(token_.getString());
           sectors <= std::numeric_limits<unsigned>::max()) {
```

Reply:
Heh, I had that at first, and then I thought to be clever to to fit it on one line :D. I’ll change it.